hikari/impl/special_endpoints.py

```diff
@@ -609,6 +609,55 @@ async def _next_chunk(self) -> typing.Optional[typing.Generator[applications.Own
         return (self._entity_factory.deserialize_own_guild(g) for g in chunk)
 
 
+# We use an explicit forward reference for this, since this breaks potential
+# circular import issues (once the file has executed, using those resources is
+# not an issue for us).
+class GuildBanIterator(iterators.BufferedLazyIterator["guilds.GuildBan"]):
+    """Iterator implementation for retrieving guild bans."""
+
+    __slots__: typing.Sequence[str] = (
+        "_entity_factory",
+        "_guild_id",
+        "_request_call",
+        "_route",
+        "_first_id",
+        "_last_id",
+    )
+
+    def __init__(
+        self,
+        entity_factory: entity_factory_.EntityFactory,
+        request_call: typing.Callable[
+            ..., typing.Coroutine[None, None, typing.Union[None, data_binding.JSONObject, data_binding.JSONArray]]
+        ],
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        first_id: undefined.UndefinedOr[snowflakes.SnowflakeishOr[users.PartialUser]],
```

Comment:
can this handle before and after like the message iterator or scheduled event iterator (having a direction and starts_at parameters), its just nicer to not allow passing both at the same time on this level